cluster-service/deploy/templates/k8s-job-arbitrary-commands.yaml

```diff
@@ -0,0 +1,103 @@
+{{ if .Values.deployDebugJob }}
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: cs-arbitrary-commands
+  namespace: '{{ .Release.Namespace }}'
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: cs-arbitrary-commands-role
+  namespace: '{{ .Release.Namespace }}'
+rules:
+  - apiGroups: [""]
+    resources: ["pods"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: ["apps"]
+    resources: ["deployments"]
+    verbs: ["get", "list", "watch"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: cs-arbitrary-commands-rolebinding
+  namespace: '{{ .Release.Namespace }}'
+subjects:
+  - kind: ServiceAccount
+    name: cs-arbitrary-commands
+    namespace: '{{ .Release.Namespace }}'
+roleRef:
+  kind: Role
+  name: cs-arbitrary-commands-role
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: cs-arbitrary-commands
+  namespace: '{{ .Release.Namespace }}'
+spec:
+  ttlSecondsAfterFinished: 300
+  activeDeadlineSeconds: 900
+  template:
+    spec:
+      serviceAccountName: cs-arbitrary-commands
+      restartPolicy: Never
+      containers:
+      - name: cs-arbitrary-commands
+        image: 'mcr.microsoft.com/aks/command/runtime:master.240118.1'
+        command: ["/bin/sh", "-c"]
+        args:
+          - |
+            checkImageTag() {
+              local EXPECTED_TAG="$1"
+              
+              # Verifying all running pods have the same image version
+              IMAGES=$(kubectl get pods -l app=clusters-service -o jsonpath='{range .items[*]}{.spec.containers[*].image}{"\n"}{end}' | sort | uniq)
+              IMAGE_COUNT=$(echo "$IMAGES" | wc -l)
+              if [ "$IMAGE_COUNT" -ne 1 ]; then
+                echo "Error: Multiple image versions detected:"
+                echo "$IMAGES"
+                exit 1
+              fi
+              
+              # Verifying image tag is the expected one
+              DEPLOYMENT_IMAGE=$(kubectl get deployment clusters-service -o jsonpath='{.spec.template.spec.containers[0].image}')
+              if [ "${DEPLOYMENT_IMAGE##*@}" != "$EXPECTED_TAG" ]; then
+                echo "Error: Deployment is using the wrong image tag"
+                echo "Current image: $DEPLOYMENT_IMAGE | Expected: $EXPECTED_TAG"
+                exit 1
+              fi
+
+              # Check if all running pods also have the correct tag
+              POD_IMAGES=$(kubectl get pods -l app=clusters-service -o jsonpath='{range .items[*]}{.spec.containers[*].image}{"\n"}{end}' | sort | uniq)
```

Comment:
Similar to https://github.com/Azure/ARO-HCP/pull/3115/files#r2472523128 let's focus on the specific container name